alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -253,6 +292,56 @@ public static function file_put_contents ( FileSyncKey $key , $content , $strict
 		self::setPermissions($fullPath);
 		self::createSyncFileForKey($rootPath, $filePath,  $key , $strict , !is_null($res), false, md5($content), kPathManager::getStorageProfileIdForKey($key));
 		self::encryptByFileSyncKey($key, kPathManager::getStorageProfileIdForKey($key));
+		self::setInSmallFileCache($key, $content);
+	}
+
+	protected static function setInSmallFileCache($key, $content)
+	{
+		$contentToPut = gzcompress($content) ??  $content;
+		$compressedContentToPut = gzcompress($contentToPut);
```

Comment:
gzcompress is called twice
if gzcompress/gzuncompress fail they return false, and it can return (without result) from get/put inSmallFileCache